core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -149,6 +156,21 @@ public BigQuerySqlDialect(SqlDialect.Context context) {
     unparseFetchUsingLimit(writer, offset, fetch);
   }
 
+  /* Unparses the given identifier with parentheses at the end if it is
+   * required to have them. */
+  @Override public void unparseSqlIdentifier(SqlWriter writer,
+      SqlIdentifier identifier, int leftPrec, int rightPrec) {
+    super.unparseSqlIdentifier(writer, identifier, leftPrec, rightPrec);
+
+    boolean isValidIdentifier = identifier.names.size() == 1;
```

Comment:
If there is an invalid identifier, will this just not unparse anything? Should we be throwing an exception or unparsing just the final part of the compound identifier instead?

Reply:
It will unparse as normal, there just won't be any parentheses.

Reply:
And sure, I'll remove the uppercase tests.

Reply:
Ah yes, I see that now. Sounds good.